Feature/OP-1258: Setup Build Scripts for Logstash

gzhou1 opened the pull request.

joelbcastillo commented on build_scripts/elk_setup/elk_setup.sh:
```diff
@@ -0,0 +1,35 @@
+mkdir -p /data/es_logs
+chown -R vagrant:vagrant /data
+chmod 777 -R /data
+# Add the following lines to /etc/security/limits.conf for max number of threads error
```
Can you use echo to write these lines to `/etc/security/limits.conf`?

Something like
```bash
# Increase the maximum number of threads ElasticSearch is allowed to use
cat << 'EOF' >> /etc/security/limits.conf
elasticsearch - nproc 2048
elasticsearch - nofile 65536
EOF
```

joelbcastillo commented on build_scripts/elk_setup/elk_setup.sh:
```diff
@@ -0,0 +1,35 @@
+# Add the following lines to /etc/security/limits.conf for max number of threads error
+# elasticsearch - nproc 2048
+# elasticsearch - nofile 65536
+service elasticsearch start
```
Don't we also need to do something so that ElasticSearch / Java uses IPv4 instead of IPv6?

gzhou1 replied:
ElasticSearch will automatically use IPv4 when supply a proper IPv4 address in the `network_host` variable in the configuration file (elasticsearch.yml). Additionally, if `ES_JAVA_OPTS` is set to `-Djava.net.preferIPv4Stack=true` in the vagrant vm, an IPv6 socket will be used for IPv4. To truly have ElasticSearch use IPv4, IPv6 needs to be disabled at the OS level.

See https://github.com/elastic/elasticsearch/issues/24591.

joelbcastillo commented:
Reviewed 1 of 14 files at r1, 13 of 14 files at r2, 1 of 7 files at r3, 1 of 1 files at r4.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[build_scripts/elk_setup/elk_setup.sh, line 17 at r1](https://reviewable.io:443/reviews/cityofnewyork/nycopenrecords/228#-KwGiwaM9R1hfAoVC11E-r1-17:-KwGkFHBNvxf44pWeWPK:bg7igkv) ([raw file](https://github.com/cityofnewyork/nycopenrecords/blob/64985beafd62ebcb7df810e0e49b3e60df3314f9/build_scripts/elk_setup/elk_setup.sh#L17)):*
<details><summary><i>Previously, joelbcastillo (Joel Castillo) wrote…</i></summary><blockquote>

Can you use echo to write these lines to `/etc/security/limits.conf`?

Something like
```bash
# Increase the maximum number of threads ElasticSearch is allowed to use
cat << 'EOF' >> /etc/security/limits.conf
elasticsearch - nproc 2048
elasticsearch - nofile 65536
EOF
```
</blockquote></details>

<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cityofnewyork/nycopenrecords/228)*
<!-- Sent from Reviewable.io -->

joelbcastillo commented:
Noticing two remaining issues:
1) Testing Logstash using `bin/logstash -e 'input { stdin { } } output { stdout {} }'` does not seem to work. We are getting an error where Logstash cannot find the logstash.yml configuration file. 
2) Possibly an issue with my machine: I am unable to access anything through the browser. Curl works, so I'm guessing it is something to do with my local machine network configuration. @johnyu95 can you test this out to make sure it works on another machine?

---

Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cityofnewyork/nycopenrecords/228#-:-KwGph_FZ2e0euTBtzdc:b-8zedzx)*
<!-- Sent from Reviewable.io -->

gzhou1 commented:
Addressing the issues:

1. Test Logstash by using `sudo /usr/share/logstash/bin/logstash --path.settings=/etc/logstash -e 'input { stdin { } } output { stdout {} }'` should work. As per https://github.com/elastic/logstash/issues/6582, the warning where Logstash cannot find logstash.yml is expected behavior and we have to pass in `path.settings` to resolve.

2. Have you tried going to https://10.0.0.4:5602 to access Kibana and https://10.0.0.4:9201 to access Elasticsearch?

joelbcastillo commented:
So when using the command listed above to test logstash you can see that it is writing to /var/log/logstash instead of an elasticsearch index. 

See https://qbox.io/blog/how-to-elasticsearch-logstash-kibana-manage-logs for an example of setting up an input and output configuration for logstash

joelbcastillo commented:
Input and output for logstash will be completed in another ticket.
